ledger/db/src/mint_config_store.rs

```diff
@@ -0,0 +1,1144 @@
+                &block_index_bytes,
+                //  ensures we do not overwrite a nonce that was already used
+                WriteFlags::NO_OVERWRITE,
+            )?;
+
+            db_transaction.put(
+                self.active_mint_configs_by_token_id,
+                &u32_to_key_bytes(mint_config_tx.prefix.token_id),
+                &encode(&active_mint_configs),
+                WriteFlags::empty(),
+            )?;
+        }
+
+        Ok(())
+    }
+
+    /// Get MintConfigTxs in a given block.
+    pub fn get_mint_config_txs_by_block_index(
+        &self,
+        block_index: u64,
+        db_transaction: &impl Transaction,
+    ) -> Result<Vec<MintConfigTx>, Error> {
+        let mint_config_tx_list: MintConfigTxList = decode(db_transaction.get(
+            self.mint_config_txs_by_block,
+            &u64_to_key_bytes(block_index),
+        )?)?;
+        Ok(mint_config_tx_list.mint_config_txs)
+    }
+
+    /// Get mint configurations for a given token.
+    pub fn get_active_mint_configs(
+        &self,
+        token_id: TokenId,
+        db_transaction: &impl Transaction,
+    ) -> Result<Vec<ActiveMintConfig>, Error> {
+        let token_id_bytes = u32_to_key_bytes(*token_id);
+        match db_transaction.get(self.active_mint_configs_by_token_id, &token_id_bytes) {
+            Ok(bytes) => {
+                let active_mint_configs: ActiveMintConfigs = decode(bytes)?;
+                Ok(active_mint_configs.configs)
+            }
+            Err(lmdb::Error::NotFound) => Ok(Vec::new()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    // Attempt to get a MintConfig that is active and is capable of minting the
+    // given amount of tokens.
+    pub fn get_active_mint_config_for_mint_tx(
+        &self,
+        mint_tx: &MintTx,
+        db_transaction: &impl Transaction,
+    ) -> Result<ActiveMintConfig, Error> {
+        let active_mint_configs =
+            self.get_active_mint_configs(TokenId::from(mint_tx.prefix.token_id), db_transaction)?;
+        let message = mint_tx.prefix.hash();
+
+        // Our default error is NotFound, in case we are unable to find a mint config
+        // that matches the mint tx. We might override it if we find one but the
+        // amount will exceed the mint limit.
+        let mut error = Error::NotFound;
+
+        for active_mint_config in active_mint_configs {
+            // See if this mint config has signed the mint tx.
+            if active_mint_config
+                .mint_config
+                .signer_set
+                .verify(&message, &mint_tx.signature)
+                .is_err()
+            {
+                continue;
+            }
+
+            // This mint config has signed the mint tx. Is it allowed to mint the given
+            // amount of tokens?
+            if let Some(new_total_minted) = active_mint_config
+                .total_minted
+                .checked_add(mint_tx.prefix.amount)
+            {
```

Comment:
IIUC we should return an Error when this `checked_add` returns `None`. Maybe also add a test case exercising that.

Reply:
Nope - the idea is that if this results in an error (a `None` value is returned) then we continue to look for a different active mint configuration. We don't want to abort execution in that case, just keep iterating.

Reply:
Ah, OK. That seems worth documenting.